### PR TITLE
Added implementation of the previous runtime api for the Workflow.

### PR DESCRIPTION
**Pull request by sagarkapare**
Jira - https://issues.cask.co/browse/CDAP-2077

### CODE REVIEW

**Comment by sagarkapare** [comment]
@sree Please take a look. Thanks!

**Comment by sagarkapare** [comment]
Release build is running here - http://builds.cask.co/browse/CDAP-RBT231-1

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/Scheduler.java [hunk 1]
```diff
@@ -78,6 +78,21 @@ public void schedule(Id.Program program, SchedulableProgramType programType, Ite
     throws SchedulerException;
 
   /**
+   * Get the previous run time for the program. A program may contain one or more schedules
+   * the method returns the previous runtimes for all the schedules. This method only takes
+   + into account {@link Schedule}s based on time. For schedules based on data, an empty list will
+   + be returned.
+   *
+   * @param program program to fetch the next runtime.
```
Comment (sreevatsanraman) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/Scheduler.java]: Previous runtime

**Comment by sreevatsanraman** [comment]
Minor comment fix and merge. LGTM

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/Scheduler.java (hunk 1 above)
Comment (sagarkapare) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/Scheduler.java]: Fixed comment.

**Comment by sagarkapare** [comment]
New build is running here - http://builds.cask.co/browse/CDAP-RBT231-2

**Comment by sagarkapare** [comment]
Unrelated test case `TestAdapterFrameworkTestRun testWorkflowAdapter` failed, which is failing on the release as well. Merging this.

**Pull request merged**